samples/nrf9160/serial_lte_modem/src/lwm2m/th_lwm2m_service.c

```diff
@@ -0,0 +1,727 @@
+		lwm2m_engine_set_res_data("3/0/6/1", &usb_idx,
+			sizeof(usb_idx), 0);
+		lwm2m_engine_create_res_inst("3/0/7/1");
+		lwm2m_engine_set_res_data("3/0/7/1", &usb_mv,
+			sizeof(usb_mv), 0);
+		lwm2m_engine_create_res_inst("3/0/8/1");
+		lwm2m_engine_set_res_data("3/0/8/1", &usb_ma,
+			sizeof(usb_ma), 0);
+	}
+
+	/* setup FIRMWARE object */
+#if defined(CONFIG_LWM2M_FIRMWARE_UPDATE_OBJ_SUPPORT)
+	/* setup data buffer for block-wise transfer */
+	lwm2m_engine_register_pre_write_callback("5/0/0", firmware_get_buf);
+	lwm2m_firmware_set_write_cb(firmware_block_received_cb);
+#endif
+#if defined(CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_SUPPORT)
+	lwm2m_firmware_set_update_cb(firmware_update_cb);
+#endif
+
+	if (!object_instances_created) {
+#if CONFIG_LWM2M_IPSO_LIGHT_CONTROL_INSTANCE_COUNT > 0
+		/* IPSO: Light Control object */
+		lwm2m_engine_register_create_callback(3311,
+			light_control_create_cb);
+		lwm2m_engine_register_delete_callback(3311,
+			light_control_delete_cb);
+		lwm2m_engine_create_obj_inst("3311/0");
+		lwm2m_engine_register_post_write_callback("3311/0/5850",
+			light_control_write_cb);
+		lwm2m_engine_set_res_data("3311/0/5750",
+			LIGHT_NAME1, sizeof(LIGHT_NAME1),
+			LWM2M_RES_DATA_FLAG_RO);
+#endif
+#if CONFIG_LWM2M_IPSO_LIGHT_CONTROL_INSTANCE_COUNT > 1
+		lwm2m_engine_create_obj_inst("3311/1");
+		lwm2m_engine_register_post_write_callback("3311/1/5850",
+			light_control_write_cb);
+		lwm2m_engine_set_res_data("3311/1/5750",
+			LIGHT_NAME2, sizeof(LIGHT_NAME2),
+			LWM2M_RES_DATA_FLAG_RO);
+#endif
+	}
+
+	/* All instances only need to create once */
+	object_instances_created = true;
+
+	return 0;
+}
+
+static void rd_client_event(struct lwm2m_ctx *client,
+	enum lwm2m_rd_client_event client_event)
+{
+	u8_t evt;
+
+	switch (client_event) {
+
+	case LWM2M_RD_CLIENT_EVENT_NONE:
+		/* do nothing */
+		break;
+
+	case LWM2M_RD_CLIENT_EVENT_BOOTSTRAP_REG_FAILURE:
+		LOG_ERR("Bootstrap registration failure!");
+		break;
+
+	case LWM2M_RD_CLIENT_EVENT_BOOTSTRAP_REG_COMPLETE:
+		LOG_INF("Bootstrap registration complete");
+		break;
+
+	case LWM2M_RD_CLIENT_EVENT_BOOTSTRAP_TRANSFER_COMPLETE:
+		LOG_INF("Bootstrap transfer complete");
+		break;
+
+	case LWM2M_RD_CLIENT_EVENT_REGISTRATION_FAILURE:
+		LOG_ERR("Registration failure!");
+		break;
+
+	case LWM2M_RD_CLIENT_EVENT_REGISTRATION_COMPLETE:
+		LOG_INF("Registration complete");
+		break;
+
+	case LWM2M_RD_CLIENT_EVENT_REG_UPDATE_FAILURE:
+		LOG_ERR("Registration update failure!");
+		break;
+
+	case LWM2M_RD_CLIENT_EVENT_REG_UPDATE_COMPLETE:
+		LOG_INF("Registration update complete");
+		break;
+
+	case LWM2M_RD_CLIENT_EVENT_DEREGISTER_FAILURE:
+		LOG_ERR("Deregister failure!");
+		break;
+
+	case LWM2M_RD_CLIENT_EVENT_DISCONNECT:
+		LOG_INF("Disconnected");
+		break;
+
+	default:
+		LOG_WRN("Unknown event");
+		break;
+	}
+
+	evt = client_event;
+	(void)inter_connect_notify(NOT_TYPE_LWM2M_RD, &evt, 1);
+}
+
+int do_lwm2m_connect(const u8_t *endpoint_name, u8_t length)
```

Comment:
Please make functions local to this file static.

Reply:
This function will be exposed to raw AT mode handling module (within the sample project) later. Please allow me to keep it as it is for now. Same to other similar cases.